From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 2 Jun 2010 08:49:16 -0400
Subject: [misc] workqueue: silence kabi checker
Message-id: <20100602104916.2c948878@dhcp-lab-109.englab.brq.redhat.com>
Patchwork-id: 25926
O-Subject: [RHEL5 PATCH 8/7] bz#596626:  workqueues: silence kabi checker
Bugzilla: 596626
RH-Acked-by: Oleg Nesterov <oleg@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Fix below ABI checker error:

silent
*** ERROR - ABI BREAKAGE WAS DETECTED ***

The following symbols have been changed (this will cause an ABI breakage):

queue_delayed_work_on
queue_delayed_work
flush_workqueue
cancel_rearming_delayed_workqueue
__create_workqueue
queue_work
destroy_workqueue

Nobody outside of workqueue.c knows about struct cpu_workqueue_struct,
thus this is a false positive, it is safe to change it and wrap the
addition with __GENKSYMS__.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/workqueue.c b/kernel/workqueue.c
index fb1fc3f..58d5669 100644
--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -45,7 +45,9 @@ struct cpu_workqueue_struct {
 
 	long remove_sequence;	/* Least-recently added (next to run) */
 	long insert_sequence;	/* Next to add */
+#ifndef __GENKSYMS__
 	struct work_struct *current_work;
+#endif
 
 	struct list_head worklist;
 	wait_queue_head_t more_work;
